app/src/main/java/com/android/bookswap/ui/profile/UserProfile.kt

```diff
@@ -0,0 +1,168 @@
+                  Box {
+                    IconButton(onClick = { /*TODO*/}, modifier = Modifier.aspectRatio(1f)) {
+                      Box(
+                          modifier =
+                              Modifier.padding(2.5f.dp)
+                                  .border(3.5f.dp, Color(0xFFA98467), CircleShape)) {
+                            Image(
+                                imageVector = Icons.Rounded.AccountCircle,
+                                contentDescription = "",
+                                modifier = Modifier.fillMaxSize().scale(1.2f).clipToBounds(),
+                                colorFilter = ColorFilter.tint(Color(0xFF6C584C)))
+                          }
+                      Box(
+                          modifier = Modifier.fillMaxSize().padding(0f.dp),
+                          contentAlignment = Alignment.TopEnd) {
+                            IconButton(onClick = { /*TODO*/}, modifier = Modifier) {
+                              Image(
+                                  imageVector = Icons.Outlined.Edit,
+                                  contentDescription = "",
+                                  colorFilter = ColorFilter.tint(Color(0xFFAAAAAA)))
+                            }
+                          }
+                    }
+                  }
+                }
+                Column(Modifier.fillMaxHeight().fillMaxWidth(), Arrangement.spacedBy(8.dp)) {
+                  // Full name text
+                  Text(
+                      text = "${user.greeting} ${user.firstName} ${user.lastName}",
+                      modifier = Modifier.testTag("fullNameTxt"))
+
+                  // Email text
+                  Text(text = user.email, modifier = Modifier.testTag("emailTxt"))
+
+                  // Phone number text
+                  Text(text = user.phoneNumber, modifier = Modifier.testTag("phoneNumberTxt"))
+
+                  // User address
+                  Text(
+                      text = "${user.latitude}, ${user.longitude}",
+                      modifier = Modifier.testTag("addressTxt"))
+
+                  // Edit Button
+                  ElevatedButton(
+                      onClick = { showEditProfile = true },
+                      border = BorderStroke(1f.dp, Color(0xFFA98467)),
+                      modifier = Modifier.testTag("editProfileBtn")) {
+                        // Text displayed on the button
+                        Text("Edit Profile")
+                      }
+                }
+              }
+        }
+  }
+}
+
+// @Preview(showBackground = true, widthDp = 540, heightDp = 1110)
+// @Composable
+// fun UserProfilePreview() {
+//  val userVM = UserViewModel("")
+//  userVM.updateUser(
+//    DataUser(
+//      "M.",
+//      "John",
+//      "Doe",
+//      "John.Doe@example.com",
+//      "+41223456789",
+//      0.0,
+//      0.0,
+//      "dummyPic.png",
+//      "dummyUUID0000")
+//  )
+//  UserProfile(userVM)
+// }
```

Comment:
Need to remove the commented code

Reply:
This is a preview used in Android Studio to make UI design easier and faster by being able to render the UI without launching the app on the emulator, however , there is no way to "test" this part so it is currently commented to avoid reducing line coverage